frontend/src/app/Settings/Facility/FacilityForm.tsx

```diff
@@ -1,191 +1,145 @@
   deviceTypes: FacilityFormDeviceType[];
-  saveFacility: (facility: Facility) => void;
+  saveFacility: (facilityFormData: FacilityFormData) => void;
   newOrg?: boolean;
 }
 
+const getDefaultValues = (facility: Facility) => {
+  return {
+    facility: {
+      name: facility.name,
+      phone: facility.phone
+        ? formatPhoneNumberParens(facility.phone) ?? ""
+        : "",
+      email: facility.email,
+      street: facility.street,
+      streetTwo: facility.streetTwo,
+      city: facility.city,
+      zipCode: facility.zipCode,
+      state: facility.state,
+      cliaNumber: facility.cliaNumber,
+    },
+    orderingProvider: {
+      firstName: facility.orderingProvider.firstName,
+      middleName: facility.orderingProvider.middleName,
+      lastName: facility.orderingProvider.lastName,
+      suffix: facility.orderingProvider.suffix,
+      NPI: facility.orderingProvider.NPI,
+      phone: facility.orderingProvider.phone
+        ? formatPhoneNumberParens(facility.orderingProvider.phone) ?? ""
+        : null,
+      street: facility.orderingProvider.street,
+      streetTwo: facility.orderingProvider.streetTwo,
+      city: facility.orderingProvider.city,
+      zipCode: facility.orderingProvider.zipCode,
+      state: facility.orderingProvider.state ?? "",
+    },
+    devices: facility.deviceTypes.length
+      ? facility.deviceTypes.map((device) => device.internalId)
+      : [],
+  };
+};
 const FacilityForm: React.FC<Props> = (props) => {
-  const [facility, updateFormData] = useState<Facility>(props.facility);
-  const [formChanged, updateFormChanged] = useState<boolean>(false);
+  const facility = props.facility;
   const [addressModalOpen, setAddressModalOpen] = useState(false);
   const [suggestedAddresses, setSuggestedAddresses] = useState<
     AddressSuggestionConfig<AddressOptions>[]
   >([]);
-
-  const updateForm: typeof updateFormData = (data) => {
-    updateFormData(data);
-    updateFormChanged(true);
-  };
-
-  const updateFacility = (newFacility: Facility) => {
-    updateForm({
-      ...facility,
-      ...newFacility,
-    });
-  };
-
-  const updateProvider = (orderingProvider: Provider) => {
-    updateForm({
-      ...facility,
-      orderingProvider,
-    });
-  };
-
-  const updateSelectedDevices = (deviceTypes: DeviceType[]) => {
-    updateForm((facility) => ({
-      ...facility,
-      deviceTypes,
-    }));
+  const {
+    watch,
+    register,
+    handleSubmit,
+    setFocus,
+    control,
+    setValue,
+    setError,
+    getValues,
+    getFieldState,
+    clearErrors,
+    formState: { errors, isSubmitting, isDirty },
+  } = useForm({
+    defaultValues: getDefaultValues(facility),
+  });
+
+  const updateSelectedDevices = (selectedItems: string[]) => {
+    setValue("devices", selectedItems, { shouldDirty: true });
+    if (selectedItems.length === 0) {
+      setError("devices", {
+        type: "required",
+        message: "There must be at least one device",
```

Comment:
similar note to the comment [here](https://github.com/CDCgov/prime-simplereport/pull/5983/files#r1235287974)